File: bridges/nodejs/src/sdk/toolbox.ts
import { INTENT_OBJECT } from '@bridge/constants'

/**
 * Get the widget id if any
 * @example getWidgetId() // 'timerwidget-5q1xlzeh
 */
export function getWidgetId(): string | null {
  return (
    INTENT_OBJECT.current_entities.find(
      (entity) => entity.entity === 'widgetid'
    )?.sourceText ?? null
  )
}

File: bridges/python/src/sdk/toolbox.py
from typing import Optional
from ..constants import INTENT_OBJECT


def get_widget_id() -> Optional[str]:
    """
    Get the widget ID if any
    """
    for entity in INTENT_OBJECT['current_entities']:
        if entity['entity'] == 'widgetid':
            return entity['sourceText']

    return None
